### PR TITLE
Improve readability of Paginator#initialize

### PR DESCRIPTION
**Pull request by ashmaroli**
I had a difficult time trying to understand what was happening within the `Paginator#initialize` method.
So made an attempt to ease the process for a future reader.

### CODE REVIEW

**Comment by sverrirs** [comment]
Ok, and now I have a difficult time to understand your changes. What is the best option here? 
This code has some test cover but not nearly enough to ensure that the rewritten code does not break basic behavior. 

This classifies as changes that bring no user benefits, I'm afraid I must decline this addition :/

**Comment by ashmaroli** [comment]
> What is the best option here?

Leverage the beauty of Ruby. Use predefined syntactic-sugar to replace verbose syntaxes and verbose syntaxes to clarify statements.
Few examples used here:

- `String#empty?` instead of `String#length >= 0`
- `attr_reader` initializes an instance_var with value `nil`. Therefore, `@ivar = loc_var == 'foo' ? 'bar' : nil` can be simplified to `@ivar = 'bar' if loc_var == 'foo'`
- `@variable = foo ? nil : ( lorem ? ipsum : dolor )` simplified to:
    ```ruby
    unless foo
      @variable = if lorem
                    return ipsum
                  else
                    return dolor
                  end
    end
    ```

**Comment by sverrirs** [comment]
OK, sounds reasonable, I'll have a better look at this tonight and brush up on my Ruby skills ;)

**Comment by sverrirs** [comment]
Ah darn I did a stupid, the PR branch has been deleted and I can't reopen this

**Comment by ashmaroli** [comment]
The branch has been restored.. Glad you're taking a second look at this..

File: lib/jekyll-paginate-v2/generator/paginator.rb [hunk 1]
```diff
@@ -20,42 +20,60 @@ def initialize(config_per_page, first_index_page_url, paginated_page_url, posts,
         @per_page = config_per_page.to_i
         @total_pages = num_pages
 
-        if @page > @total_pages
-          raise RuntimeError, "page number can't be greater than total pages: #{@page} > #{@total_pages}"
-        end
+        # RuntimeError is raised by default
+        raise "page number can't be greater than total pages: #{@page} > #{@total_pages}" if @page > @total_pages
```
Comment (sverrirs) [inline lib/jekyll-paginate-v2/generator/paginator.rb]: The placement of conditions at the end of lines hurts readability, it forces the dev to evaluate the entire line for the conditions. Please don't do this

File: lib/jekyll-paginate-v2/generator/paginator.rb [hunk 2]
```diff
@@ -20,42 +20,60 @@ def initialize(config_per_page, first_index_page_url, paginated_page_url, posts,
...
-          # Adjust the paginated pages as well
-          paginated_page_url = Utils.ensure_full_path(paginated_page_url, default_indexpage, default_ext)
+          paginated_page_url   = Utils.ensure_full_path(paginated_page_url, default_indexpage, default_ext)
         end        
 
         @total_posts = posts.size
-        @posts = posts[init..offset]
-        @page_path = Utils.format_page_number(this_page_url, cur_page_nr, @total_pages)
+        @posts       = posts[init..offset]
+        @page_path   = _number_page(this_page_url, cur_page_nr)
```
Comment (sverrirs) [inline lib/jekyll-paginate-v2/generator/paginator.rb]: Please retain the name of the Utils function if you want to introduce a wrapper, eg `_format_page_number` or alternatively keep the part of the name that describes the best what its purpose is, in this case I'd be happy with `_format`

File: lib/jekyll-paginate-v2/generator/paginator.rb [hunk 3]
```diff
@@ -20,42 +20,60 @@ def initialize(config_per_page, first_index_page_url, paginated_page_url, posts,
...
-        @previous_page = @page != 1 ? @page - 1 : nil
-        @previous_page_path = @page == 1 ? nil : 
-                              @page == 2 ? Utils.format_page_number(first_index_page_url, 1, @total_pages) : 
-                              Utils.format_page_number(paginated_page_url, @previous_page, @total_pages)
-        @next_page = @page != @total_pages ? @page + 1 : nil
-        @next_page_path = @page != @total_pages ? Utils.format_page_number(paginated_page_url, @next_page, @total_pages) : nil
+        # First page has no "previous page"
+        unless @page == 1
+          @previous_page = @page - 1
+          @previous_page_path = if @page == 2
```
Comment (sverrirs) [inline lib/jekyll-paginate-v2/generator/paginator.rb]: In this case a ternary operator would be clearer

File: lib/jekyll-paginate-v2/generator/paginator.rb (hunk 1 above)
Comment (ashmaroli) [reply lib/jekyll-paginate-v2/generator/paginator.rb]: Fixed..

File: lib/jekyll-paginate-v2/generator/paginator.rb (hunk 2 above)
Comment (ashmaroli) [reply lib/jekyll-paginate-v2/generator/paginator.rb]: renamed to `_format_page_number`

File: lib/jekyll-paginate-v2/generator/paginator.rb (hunk 3 above)
Comment (ashmaroli) [reply lib/jekyll-paginate-v2/generator/paginator.rb]: Wouldn't that force a dev to evaluate the entire line (which is very long) :wink:
Multi-line ternary operations are harder to read IMO..

**Comment by ashmaroli** [comment]
I split up part of the logic into a another private method to reduce the "Cognitive Complexity" score reported by CodeClimate..

File: lib/jekyll-paginate-v2/generator/paginator.rb (hunk 3 above)
Comment (ashmaroli) [reply lib/jekyll-paginate-v2/generator/paginator.rb]: I had a hunch that line could be made more DRY.. but just couldn't get there..
Applying your suggestion with a local variable..

**Pull request closed without merging**